Pod/Classes/SeafConnection.m

```diff
@@ -1931,4 +1944,52 @@ - (void)clearAccountCache
     [[NSNotificationCenter defaultCenter] postNotificationName:@"clearCache" object:nil];
     [self clearUploadCache];
 }
+
+// fileProvider tagData
+- (void)saveFileProviderTagData:(NSData*)tagData withItemIdentifier:(NSString*)itemId {
+    NSMutableDictionary *dict = [NSMutableDictionary dictionaryWithDictionary:[SeafStorage.sharedObject objectForKey:self.tagDataKey]];
+    if (tagData) {
+        [dict setObject:tagData forKey:itemId];
+    } else {
+        [dict removeObjectForKey:itemId];
+    }
+    
+    [SeafStorage.sharedObject setObject:dict forKey:self.tagDataKey];
+    // Save to iCloud
+    NSUbiquitousKeyValueStore *store = [NSUbiquitousKeyValueStore defaultStore];
```

Comment:
it might takes a long time to save to icloud, this should be done in background so that it won't block UI